### PR TITLE
[s] Fixed admins being able to bypass proccall protections via remote sdql and circuits.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

Admins can bypass IsAdminAdvancedProcCall checks by using these methods of proccall because proccall protection is kinda dumb. This has been tweaked so that there is proper proccall protection for these methods of proccall. 
Code is hacky, but there's not much of a choice if we want procs to be properly protected from admin proccalls from any sort of remote source. If anyone has a better idea on how to implement this, feel free to hit me up.

We need a special global mob that handles proccalls from sources that may not have a usr/client to refer back to. `IsAdminAdvancedProcCall()` relies usr being defined, so if no usr is defined, then this will always return false. This has been adjusted so that proccalls without a usr/client to refer back to will instead set usr to this special mob, which will then let the `IsAdminAdvancedProcCall()` return true by comparing whether usr == this special global mob.

## Why It's Good For The Game

Admins can no longer bypass IsAdminAdvancedProcCall checks.

## Changelog

:cl:
admin: Admins are no longer able to bypass proccall protections using remote methods of proccalling.
/:cl:

### CODE REVIEW

**Comment by MrStonedOne** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current ~~title and or~~ body content does not meet requirements, so please update it.

File: code/modules/admin/callproc/callproc.dm [hunk 1]
```diff
@@ -1,3 +1,94 @@
+
+GLOBAL_DATUM_INIT(AdminProcCallHandler, /mob/proccall_handler, new())
+GLOBAL_PROTECT(AdminProcCallHandler)
+
+/// Used to handle proccalls called indirectly by an admin (e.g. tgs, circuits)
+/mob/proccall_handler
+	name = "ProcCall Handler"
+	desc = "If you are seeing this, tell a coder."
```
Comment (MrStonedOne) [inline code/modules/admin/callproc/callproc.dm:8]: autodoc should explain why this is a mob and not anything else.

File: code/modules/admin/callproc/callproc.dm [hunk 2]
```diff
@@ -1,3 +1,96 @@
+
+GLOBAL_DATUM_INIT(AdminProcCallHandler, /mob/proccall_handler, new())
+GLOBAL_PROTECT(AdminProcCallHandler)
+
+/// Used to handle proccalls called indirectly by an admin (e.g. tgs, circuits).
+/// Has to be a mob because IsAdminAdvancedProcCall() checks usr, which is a mob variable.
+/// So usr is set to this for any proccalls that don't have any usr mob/client to refer to.
```
Comment (Fikou) [inline code/modules/admin/callproc/callproc.dm:7]: dont you use /** for these kinds of autodoc comments

Comment (Watermelon914) [reply code/modules/admin/callproc/callproc.dm:7]: I think you can do it both ways. I've seen mothblocks do it like this

**Comment by optimumtact** [comment]
Personally I don't think this has ever been worth the cost, unless we have proper RBAC on proc calling this is pointless

File: code/modules/admin/callproc/callproc.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/admin/callproc/callproc.dm:7]: i always prefer multiple `///` just as a side effect of programming a lot in rust

**Pull request merged**